packages/remix/src/utils/futureFlags.ts

```diff
@@ -33,3 +40,26 @@ export function getFutureFlagsBrowser(): FutureConfig | undefined {
 export function getFutureFlagsServer(build: ServerBuild): FutureConfig | undefined {
   return build.future;
 }
+
+/**
+ * Read Remix version from module package.json
+ *
+ * @returns The major version number
+ */
+export function getRemixVersionFromPkg(): number | undefined {
+  const pkg = loadModule<{ version: string }>('@remix-run/react/package.json');
+  const version = pkg ? pkg.version : '0.0.0';
```

Comment:
m: If we stick with this approach (see my other comment), let's add a debug log if we fail to load `package.json`